micropub.php

```diff
@@ -593,15 +593,31 @@ public static function generate_post_content( $args ) {
 			'like-of'     => 'Likes',
 			'repost-of'   => 'Reposted',
 			'in-reply-to' => 'In reply to',
+			'bookmark-of' => 'Bookmarked',
 		);
 
 		// interactions
 		foreach ( array_keys( $verbs ) as $prop ) {
-			$val = $props[ $prop ][0];
+			if ( wp_is_numeric_array( $props[ $prop ] ) ) {
+				$val = $props[ $prop ][0];
+			} else {
+				$val = $props[ $prop ];
+			}
 			if ( $val ) {
-				$lines[] = '<p>' . $verbs[ $prop ] .
-				  ' <a class="u-' . $prop . '" href="' .
-				  $val . '">' . $val . '</a>.</p>';
+				$l = '<p>' . $verbs[ $prop ] .
+				' <a class="u-' . $prop . '" href="';
+				if ( is_string( $val ) ) {
+					$l .= $val . '">' . $val;
```

Comment:
the closing `>` should probably be added outside all of these conditionals. otherwise, if `$val` is something other than a string or array, or if `$val['url']` isn't set, it won't get added at all right now.

(speaking of which, that would be a good test to add. :P)

...or actually, something like `sprintf` would probably be a lot more readable and understandable than all these conditionals. mind trying that instead?

Reply:
Happy to do a sprintf. Would definitely make things a bit easier.